File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutboxSenderManager.cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Options;
using Volo.Abp.BackgroundWorkers;
using Volo.Abp.EventBus.Distributed;

namespace Volo.Abp.EventBus.Boxes;

public class OutboxSenderManager : IBackgroundWorker
{
    protected AbpDistributedEventBusOptions Options;//  { get; }
    protected IServiceProvider ServiceProvider;//  { get; }
    protected List<IOutboxSender> Senders;//  { get; }

    public OutboxSenderManager(
        IOptions<AbpDistributedEventBusOptions> options,
        IServiceProvider serviceProvider)
    {
        ServiceProvider = serviceProvider;
        Options = options.Value;
        Senders = new List<IOutboxSender>();
    }

    public void StartAsync(CancellationToken cancellationToken = default)
    {
        for (var outboxConfig in Options.Outboxes.Values)
        {
            if (outboxConfig.IsSendingEnabled)
            {
                var sender = ServiceProvider.GetRequiredService<IOutboxSender>();
                sender.StartAsync(outboxConfig, cancellationToken);
                Senders.Add(sender);
            }
        }
    }

    public void StopAsync(CancellationToken cancellationToken = default)
    {
        for (var sender in Senders)
        {
            sender.StopAsync(cancellationToken);
        }
    }
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutboxSenderManager.java
package com.ueueo.eventbus.distributed;

import com.ueueo.backgroundworkers.IBackgroundWorker;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Getter
public class OutboxSenderManager implements IBackgroundWorker {
    protected AbpDistributedEventBusOptions options;
    protected IOutboxSender outboxSender;
    protected List<IOutboxSender> senders;

    public OutboxSenderManager(
            AbpDistributedEventBusOptions options,
            IOutboxSender outboxSender) {
        this.outboxSender = outboxSender;
        this.options = options;
        this.senders = new ArrayList<>();
    }

    @Override
    public void start() {
        for (OutboxConfig outboxConfig : options.getOutboxes().values()) {
            if (outboxConfig.isSendingEnabled()) {
                outboxSender.start(outboxConfig);
                senders.add(outboxSender);
            }
        }
    }

    @Override
    public void stop() {
        for (IOutboxSender sender : senders) {
            sender.stop();
        }
    }

}
